tests/std/tests/P0896R4_ranges_alg_copy_n/test.cpp

```diff
@@ -9,34 +9,30 @@
 
 #include <range_algorithm_support.hpp>
 
-constexpr void smoke_test() {
-    using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
-    using std::same_as;
+using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
+using std::same_as;
 
-    // Validate that copy_n_result aliases in_out_result
-    STATIC_ASSERT(same_as<copy_n_result<int, double>, ranges::in_out_result<int, double>>);
-
-    int const input[] = {13, 42, 1729};
-    int output[]      = {-1, -1, -1};
-    basic_borrowed_range wrapped_input{input};
-    auto result = copy_n(wrapped_input.begin(), ranges::distance(input), basic_borrowed_range{output}.begin());
-    STATIC_ASSERT(same_as<decltype(result),
-        copy_n_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-    assert(result.in == wrapped_input.end());
-    assert(result.out == basic_borrowed_range{output}.end());
-    assert(ranges::equal(output, input));
-}
-
-int main() {
-    STATIC_ASSERT((smoke_test(), true));
-    smoke_test();
-}
+// Validate that copy_n_result aliases in_out_result
+STATIC_ASSERT(same_as<copy_n_result<int, double>, ranges::in_out_result<int, double>>);
 
 struct instantiator {
-    template <class In, class, class Out>
-    static void call(In in = {}, std::iter_difference_t<In> const count = 42, Out out = {}) {
-        (void) ranges::copy_n(std::move(in), count, std::move(out));
+    static constexpr int input[3] = {13, 42, 1729};
+    template <class In, class, class Write>

```

Comment:
We should ditch this middle parameter, and in `range_algorithm_support.hpp` remove `with_difference` (no caller uses the difference type), rename `test_counted_write` to `test_read_write`, rename `with_input_iterators` to `with_readable_iterators`. We'll also have to update the `for_each_n` test to call a new `test_read` that uses only `with_readable_iterators`.

Reply:
Consider renaming `In` to `Read` to be consistent with the `test_read_write` renaming, so it's clear this doesn't name a model of `input_range` as `In` does in most range alg tests.

Reply:
Ditto I should do this in #959.